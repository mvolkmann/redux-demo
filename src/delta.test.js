// @flow

import React from 'react';
import Delta from './delta';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import Enzyme, {mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({adapter: new Adapter()});

/**
 * These tests just verify that when the user
 * interacts with the UI in a certain way,
 * the expected Redux actions are dispatched.
 */
describe('Delta', () => {
  let store;

  beforeEach(() => {
    const mockStore = configureStore();
    const initialState = {delta: 1};
    store = mockStore(initialState);
  });

  test('should match snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Delta />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should dispatch', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Delta />
      </Provider>
    );
    const input = wrapper.find('input');
    input.simulate('change', {target: {value: 2}});

    const actions = store.getActions();
    expect(actions[0]).toEqual({
      type: 'deltaChange',
      payload: 2
    });
  });
});
