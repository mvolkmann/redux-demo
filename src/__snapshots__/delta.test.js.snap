// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delta should match snapshot 1`] = `
<div
  className="delta"
>
  <label>
    Delta
  </label>
  <input
    onChange={[Function]}
    type="number"
    value={1}
  />
</div>
`;
